src/test/java/com/google/neighborgood/UserInfoServletTest.java

```diff
@@ -0,0 +1,422 @@
+    Entity dummy_2 = new Entity("UserInfo");
+    dummy_2.setProperty("nickname", "Leonar");
+    dummy_2.setProperty("address", "xxx");
+    dummy_2.setProperty("phone", "xxx");
+    dummy_2.setProperty("email", "test2@example.com");
+    dummy_2.setProperty("userId", "12345");
+    dummy_2.setProperty("points", 50);
+    ds.put(dummy_2);
+
+    when(request.getParameter("nickname-input")).thenReturn("Leonard");
+    when(request.getParameter("address-input")).thenReturn("4xxx Centre Avenue");
+    when(request.getParameter("phone-input")).thenReturn("4xxxxxxxxx");
+
+    try {
+      new UserInfoServlet().doPost(request, response);
+    } catch (IOException e) {
+      assertTrue(false);
+    }
+
+    // After sending the POST request, there should be three entities in the datastore
+    assertEquals(3, ds.prepare(new Query("UserInfo")).countEntities(withLimit(10)));
+
+    // Filter out the entity that has the userId of 1234567890
+    PreparedQuery results =
+        ds.prepare(
+            new Query("UserInfo")
+                .setFilter(new FilterPredicate("userId", FilterOperator.EQUAL, "1234567890")));
+    Entity entity = results.asSingleEntity();
+    if (entity == null) {
+      // The entity can't be null
+      assertTrue(false);
+    }
+
+    // Test the stored personal information
+    assertEquals("Leonard", (String) entity.getProperty("nickname"));
+    assertEquals("4xxx Centre Avenue", (String) entity.getProperty("address"));
+    assertEquals("4xxxxxxxxx", (String) entity.getProperty("phone"));
+    assertEquals("leonardzhang@google.com", (String) entity.getProperty("email"));
+    assertEquals("1234567890", (String) entity.getProperty("userId"));
+    assertEquals(0, (long) entity.getProperty("points"));
+
+    when(request.getParameter("nickname-input")).thenReturn("Leo");
+
+    try {
+      new UserInfoServlet().doPost(request, response);
+    } catch (IOException e) {
+      assertTrue(false);
+    }
+
+    // After sending the second POST request, there should be still three entities in the datastore
+    assertEquals(3, ds.prepare(new Query("UserInfo")).countEntities(withLimit(10)));
+    results =
+        ds.prepare(
+            new Query("UserInfo")
+                .setFilter(new FilterPredicate("userId", FilterOperator.EQUAL, "1234567890")));
+    entity = results.asSingleEntity();
+    if (entity == null) {
+      // The entity can't be null
+      assertTrue(false);
+    }
+
+    // Test the stored personal information
+    assertEquals("Leo", (String) entity.getProperty("nickname"));
+    assertEquals("4xxx Centre Avenue", (String) entity.getProperty("address"));
+    assertEquals("4xxxxxxxxx", (String) entity.getProperty("phone"));
+    assertEquals("leonardzhang@google.com", (String) entity.getProperty("email"));
+    assertEquals("1234567890", (String) entity.getProperty("userId"));
+    assertEquals(0, (long) entity.getProperty("points"));
+  }
+
+  @Test
+  public void edgeCaseTest() {
```

Comment:
Can the method name be more descriptive about what edge case(s) you're testing here? If it's multiple types of edge cases, it's better to give them all their own test method.

Reply:
It is mainly a test for the edge case that the input field is empty or just spaces. I change the function name to `emptyInputTest()` in the new commit and add a comment to illustrate its purpose.